blog/posts/views.py

```diff
@@ -0,0 +1,105 @@
+import json
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from .models import PostModel 
+from django.views.generic import View
+from django.utils.decorators import method_decorator
+from users.models import UserModel
+from users.views import isAuthenticated
+
+@method_decorator(csrf_exempt, name='dispatch')
+class PostHandler(View):
+    def post(self, request):
+        response = {}
+        post_data = json.loads(request.body.decode('utf-8'))
+
+        if post_data is not None:
+            username = post_data.get('username')
+            user = UserModel.objects.filter(username=username).first() 
+            if user is not None:
+                if isAuthenticated(request, username):
+                    post_title = post_data.get('title')
+                    post = PostModel.objects.filter(author = user, title=post_title).first()
+
+                    if post is not None:
+                        response['message'] = 'Post already exists'
+                        response['title'] = post.title
+                        response['post'] = post.content
+                        return JsonResponse(response)
+
+                    new_post = PostModel(author = user, title = post_title, content = post_data.get('content'))
+                    if new_post:
+                        new_post.save()
+                        response['message'] = 'success, post created'
+                        return JsonResponse(response, status=200)
+                    else:
+                        response['error'] = 'No content'
+                        return JsonResponse(response, status=400)
+                else:
+                    return JsonResponse({'message' : f'{username} not logged in, please log in'})
+            else:
+                return JsonResponse({'message' : f'{username} not exists'})
+        else:
+            return JsonResponse({'message' : 'Invalid post'})
```

Comment:
This function has too many indents. (cyclomatic complexity). 

You can reduce it greatly by 
1) using a library that automates most of this (django rest framework)
or 
2) 
you can rewrite the function this way. 

```
if post_data is None:
   return  "Invalid Post"

if not authenticated: 
   return "not logged in"

if user is None: 
    return "user not exists"


.... 

process the request.

```

This makes the code lot easier to read.